images/base/files/etc/systemd/system/kubelet.service.d/10-kubeadm.conf

```diff
@@ -8,5 +8,8 @@ EnvironmentFile=-/var/lib/kubelet/kubeadm-flags.env
 # This is a file that the user can use for overrides of the kubelet args as a last resort. Preferably, the user should use
 # the .NodeRegistration.KubeletExtraArgs object in the configuration files instead. KUBELET_EXTRA_ARGS should be sourced from this file.
 EnvironmentFile=-/etc/default/kubelet
+# On cgroup v1, the /kubelet cgroup is created in the entrypoint script before running systemd.
+# On cgroup v2, the /kubelet cgroup is created here. (See the comments in the entrypoint script for the reason.)
+ExecStartPre=/bin/sh -euc "if [ -f /sys/fs/cgroup/cgroup.controllers ]; then create-kubelet-cgroup-v2; fi"
```

Comment:
is this script re-entrant? it tentatively looks like it is not.
under kubeadm kubelet will crash and restart many times before the config is written and it finally starts

Reply:
Only the first execution of `create-kubelet-cgroup-v2` is meaningful. So, I don't see any issue about restarting.